src/sentry/utils/encryption.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import
+
+from collections import OrderedDict
+from django.conf import settings
+from django.utils.encoding import smart_bytes
+
+
+class EncryptionManager(object):
+    def __init__(self, schemes=()):
+        for key, value in schemes:
+            if not isinstance(key, basestring):
+                raise ValueError('Encryption scheme type must be a string. Value was: {!r}'.format(value))
+            if not hasattr(value, 'encrypt') and not hasattr(value, 'decrypt'):
+                raise ValueError('Encryption scheme value must have \'encrypt\' and \'decrypt\' callables. Value was: {!r}'.format(value))
+        self.schemes = OrderedDict(schemes)
+        if not schemes:
+            self.default_scheme = None
+        else:
+            self.default_scheme = schemes[0][0]
+
+    def encrypt(self, value):
+        if self.default_scheme is None:
+            return value
+        value = smart_bytes(value)
+        scheme = self.schemes[self.default_scheme]
+        return b'{}${}'.format(self.default_scheme, scheme.encrypt(value))
+
+    def decrypt(self, value):
+        try:
+            enc_method, enc_data = value.split('$', 1)
```

Comment:
When you say values you just mean the `enc_method` name right? That seems fine to me, maybe we could blow up in `__init__` if any scheme keys us a `$`?